contractcourt/chain_watcher.go

```diff
@@ -240,6 +240,30 @@ func (c *chainWatcher) SubscribeChannelEvents(syncDispatch bool) *ChainEventSubs
 
 	if syncDispatch {
 		sub.ProcessACK = make(chan error, 1)
+
+		// If this client is syncDispatch, we cannot safely delete it
+		// from our list of clients. This is because of a potential
+		// race at shutdown, where the client shuts down and calls
+		// Cancel(). In this case we must make sure the ChainWatcher
+		// doesn't think it has successfully handed off a contract
+		// breach to the client. We start a goroutine that will send an
+		// error on the ProcessACK channel until the ChainWatcher is
+		// shutdown.
+		sub.Cancel = func() {
+			c.wg.Add(1)
+			go func() {
+				defer c.wg.Done()
+
+				err := fmt.Errorf("cancelled")
+				for {
+					select {
+					case sub.ProcessACK <- err:
```

Comment:
Not sure if there's a better way of doing this 🤔It should only happen in the very specific shutdown case, but the problem I see with doing it this way is that since it won't be deleted from the list of clients, it can still end up getting events on it's channels. Since the `closeObserver` will exit after sending one event, this shouldn't really be a problem, but it just doesn't feel right...